pkg/messages/treatyMessage.go

```diff
@@ -0,0 +1,116 @@
+package messages
+
+import (
+	"github.com/google/uuid"
+)
+
+type ConditionType int // type of condition
+type RequestType int
+type Op int
+
+const (
+	HP ConditionType = iota + 1
+	Floor
+	AvailableFood
+)
+
+const (
+	LeaveAmountFood RequestType = iota + 1
+	LeavePercentFood
+	Inform
+)
+
+const (
+	GT Op = iota + 1
+	GE
+	EQ
+	LE
+	LT
+)
+
+type Treaty struct {
+	condition      ConditionType
+	request        RequestType
+	conditionOp    Op
+	requestOp      Op
+	signatureCount int
+	duration       int
+	id             string
+}
+
+type Treatyer interface {
+	Condition() ConditionType
+	Request() RequestType
+	ConditionOp() Op
+	RequestOp() Op
+	SignatureCount() int
+	Duration() int
+	Id() string
+}
+
+func NewTreaty(condition ConditionType, request RequestType, cop Op, rop Op, duration int) *Treaty {
+	treaty := &Treaty{
+		condition:      condition,
+		request:        request,
+		conditionOp:    cop,
+		requestOp:      rop,
+		signatureCount: 0,
```

Comment:
```suggestion
		signatureCount: 1,
```

If we're proposing a treaty, we've implicitly already signed it